packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -1061,8 +1132,17 @@ class _RenderSegmentedControl<T extends Object> extends RenderBox
 
       final Rect unscaledThumbRect = state.thumbAnimatable?.evaluate(state.thumbController) ?? newThumbRect;
       currentThumbRect = unscaledThumbRect;
+
+      double delta = 0;
+      if (isLeadingChild) {
```

Comment:
what if it's both leading and trailing?

Reply:
Ah the same explanation as above:) This should mean leftmost and rightmost ones

Reply:
Oh I didn't realize `assert(children.length >= 2)`. Nvm.

Reply:
But in that case they're logically mutually exclusive. Maybe use an enum to model that instead of 2 bools.

Reply:
Just pushed a commit, does this look okay? https://github.com/flutter/flutter/pull/152976/commits/51277b6c615b0f17edbc81725ebc56a80d2c0535